### PR TITLE
acpica: update to 20180105

### PR DESCRIPTION
**Pull request by htot**
This is part of the work to enable acpi for Edison. Andy Shevshenko's latest u-boot with acpi support for Edison requires iasl greater 20170531, but Ycoto provides iasl 20160527 or acpica 20170303. I took the pyro version including YP patches, updated PV to the latest and fixed up the YP patches that broke. I tested this on Edison by building u-boot only (only iasl).

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

### CODE REVIEW

**Review by alext-mkrs — APPROVED** [review]
Looks good to me by inspection, do you want me to do a build and live test?

EDIT: and I've just realized that's Mika's repo you want to merge into, not yours :) Anyway, if either you or Mika want a hand with testing the thing on one more system - let me know.

**Comment by htot** [comment]
@alext-mkrs the YP patches did not apply cleanly after the update, I needed to manually remove conflicts. Now, it works for building the acpi enabled version of u-boot (I think, because I only tested u-boot yet with Andy's 4.15.0-rc6 non-acpi kernel). I think @westeri meta-acpi layer would need testing against the supported boards (Joule etc), but I don't have those boards. So if you can, that would be great.

**Comment by alext-mkrs** [comment]
I see. I unfortunately don't have Joule either, but I do have Galileo gen 1 and 2 and Edisons with both Arduino and mini-breakout boards, so if testing either of them would be helpful, let me know.

**Comment by andy-shev** [comment]
The idea is good and indeed we better get latest ACPICA in the tree. What I didn't get why you removed `-Werror`. It means there are some warnings that should be mitigated?

**Comment by andy-shev** [comment]
@alext-mkrs 
Galileo Gen 2 is a one of the best choice to test Mika's repo, by the way. It has internal and external GPIOs, descrete based pin control and some perpheral to play with. So, it would be definitely great to see your tests!

**Comment by htot** [comment]
@andy-shev I just brainlessly copied that from the original recipe here http://cgit.openembedded.org/openembedded-core/tree/meta/recipes-extended/acpica/files/no-werror.patch?h=pyro and assumed the comment still holds (tools generating C that is not ANSI C). But if we can drop this patch, better.

File: recipes-bsp/acpica/files/manipulate-fds-instead-of-FILE.patch [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+-            /*
+-             * A problem with freopen is that on error, we no longer
+-             * have stderr and cannot emit normal error messages.
+-             * Emit error to stdout, close files, and exit.
+-             */
+-            fprintf (stdout,
+-                "\nCould not open debug output file: %s\n\n", Filename);
+-
+-            CmCleanupAndExit ();
+-            exit (1);
```
Comment (htot) [inline recipes-bsp/acpica/files/manipulate-fds-instead-of-FILE.patch]: I am mostly worried about this chunk, as I got the conflict here. I selected the solution from the old acpica patch, but clearly upstream had also been making changes in the same area to fix the same issue, so maybe this chunk should have been dropped.

Comment (andy-shev) [reply recipes-bsp/acpica/files/manipulate-fds-instead-of-FILE.patch]: Can we just try new ACPICA version w/o any additional 3rd party patches?

**Comment by htot** [comment]
I would say: why not. What can be expected worst case is that we get spurious warnings or possible errors don't go to stderr or something like that.The 3rd patch 'rename-yy_scan_string-manually' applied without complaints and was used to fix an issue with flex 2.6.3. However, pyro/rocko/master use 2.6.0. I will try that hopefully this evening.

**Comment by htot** [comment]
I removed the 3 patches one by one and didn't observe any change in the log.do_compile of either acpica or u-boot (I am building u-boot for 32bits target):
```
include conf/machine/include/tune-core2.inc
TUNE_CCARGS .= " -mstackrealign"
```
Finally I built -c cleansstate u-boot, to be sure, no errors.

Not tested: compiling asl that should generate an error.

I put removing the patches as a separate commit in case we run into trouble when testing and want to revert.

I'll push my acpi branch of meta-intel-edison soon

**Comment by alext-mkrs** [comment]
@andy-shev, understood, sure - I'll give it a try on Galileo gen2. Not sure I'll be able to do that soon enough for this specific PR, as I'll need to probably spin up my Galileo image build environment - or would the latest official kernel, which is something quite old, from 3.10 line, IIRC, work? In any case I'll plan this for the upcoming weekend and I'll report back when I have something meaningful.

**Comment by andy-shev** [comment]
@htot 
Whenever you get done a clean PR, please notify. I won't accept anything with reverts.

@alext-mkrs 
meta-acpi is a top layer which provides everything one needs (okay, kernel might be a bit outdated, though it's v4.1x IIRC). So, just follow the instructions in README of this project.

**Comment by htot** [comment]
@andy-shev agreed I will clean up after @alext-mkrs tested.

**Comment by alext-mkrs** [comment]
Ok, so I've created a build setup based on `meta-acpi` and dependencies and looks like it will take me a bit to get through the initial build, as I'm building this on a VM on a laptop and it's far from being done, going for a couple of hours already (typically it takes it  ~5hrs to build Edison image from scratch). In the meanwhile I've looked into this PR in more detail and indeed it looks like none of the patches is actually relevant.

On top of that, I'd recommend using the `acpica` recipe from `openembedded-core:master` instead of `pyro` - they aren't too different, but the former has a couple of fixes for an apparent typo and a deprecated variable.

BTW, @htot, are you building namely a `meta-acpi` + `meta-intel` + `poky` layer set, as the `README` describes, or you do this within the `meta-intel-edison`-based setup, with `meta-acpi` just added on top?

**Comment by htot** [comment]
@alext-mkrs I'm building native, but on btrfs with hdd takes 8 hours. yesterday got a sdd for this and reduced to 2.5 hours. Also flashing sd card takes 15 min, now booting from usb3 stick, flashing takes now 1-2 min.

I put meta-intel under poky and meta-acpi next to meta-intel-edison as we're working on that. Added both layers, but don't know what the dependency on meta-intel is. Then added acpica-native as dependency to u-boot recipe. I hope to push my stuff as a proposed branch soon, but now spend time building and testing u-boot w/wo acpi with 4.15-rc7 kernels w/wo acpi combinations.

**Comment by alext-mkrs** [comment]
Understood, thanks. Yeah, I have an SSD in that laptop and that makes a huge difference.

I've created a `README`-based setup for testing this one, i.e. not a `meta-intel-edison`-based, but in the context of this PR that IMHO makes more sense.

**Comment by alext-mkrs** [comment]
Ok, so I've tested this to build ok without any errors (and generate `acpi-tables.cpio`) using the following setup (based on the `README`):
```shell
BB_VERSION        = "1.30.0"
BUILD_SYS         = "x86_64-linux"
NATIVELSBSTRING   = "Ubuntu-16.04"
TARGET_SYS        = "i586-poky-linux"
MACHINE           = "galileo"
DISTRO            = "poky"
DISTRO_VERSION    = "2.1.3"
TUNE_FEATURES     = "m32 i586-nlp"
TARGET_FPU        = ""
meta              
meta-poky         
meta-yocto-bsp    = "krogoth:bddb60b101df0b2a26a9dcb1202e95f46835a731"
meta-intel        = "krogoth:1f8dd1b00ce9c72d73583c405ec392690d9b08b7"
meta-acpi         = "pr5-acpica-update:95c2bdc281bceee3c72d221bb362d3b9580a44f4"
```
I however updated the `acpica` recipe using [the latest oe-core one](https://github.com/openembedded/openembedded-core/blob/master/meta/recipes-extended/acpica/acpica_20170303.bb) as suggested previously, the difference is below (minor) and I'd suggest updating the PR using that. I've not used any of the patches @htot first added, but then removed from the PR - none of them seems to be relevant for our use cases.

```diff
diff --git a/recipes-bsp/acpica/acpica_20180105.bb b/recipes-bsp/acpica/acpica_20180105.bb
index 3f60b43..94cead6 100644
--- a/recipes-bsp/acpica/acpica_20180105.bb
+++ b/recipes-bsp/acpica/acpica_20180105.bb
@@ -24,6 +24,11 @@ UPSTREAM_CHECK_URI = "https://acpica.org/downloads"
 
 S = "${WORKDIR}/acpica-unix2-${PV}"
 
+inherit update-alternatives
+
+ALTERNATIVE_PRIORITY = "100"
+ALTERNATIVE_${PN} = "acpixtract"
+
 EXTRA_OEMAKE = "CC='${CC}' 'OPT_CFLAGS=-Wall'"
 
 do_install() {
@@ -40,7 +45,6 @@ do_install() {
 PROVIDES = "iasl"
 RPROVIDES_${PN} += "iasl"
 RREPLACES_${PN} += "iasl"
-RCONFLIGHTS_${PN} += "iasl"
+RCONFLICTS_${PN} += "iasl"
 
-NATIVE_INSTALL_WORKS = "1"
 BBCLASSEXTEND = "native"
```

**Comment by andy-shev** [comment]
Thanks! Can you submit a clean PR?

**Comment by alext-mkrs** [comment]
Sure, I simply assumed @htot would just want to update his one  as the original autho (per his latest comment), but if for whatever reason that doesn't happen until Friday, I'll submit one.

**Comment by htot** [comment]
@alext-mkrs if you send the PR that's fine with me. Whatever works! Thanks for testing!

**Comment by andy-shev** [comment]
I'm about to close this based on comments from @htot  and @alext-mkrs.

**Comment by alext-mkrs** [comment]
Sure, no problem, I'll get the PR in tomorrow (most probably) or over the weekend the latest.

**Pull request closed without merging**